src/components/common/StaffPayment.vue

```diff
@@ -0,0 +1,152 @@
+<template>
+  <v-card flat id="AR-step-5-container">
+    <div class="payment-container">
+      <label>Payment</label>
+      <v-radio-group
+        class="value mt-0 pt-0"
+        :value="paymentOption"
+        @change="onPaymentOptionChanged($event)"
+      >
+        <!-- Payment Received radio button -->
+        <v-radio id="payment-received-radio" label="Payment Received" />
+        <v-form class="ml-8" ref="form" v-model="formValid">
+          <v-text-field
+            filled
+            persistent-hint
+            id="routing-slip-number-textfield"
+            label="Enter the Routing Slip Number "
+            hint="Fee Accounting System Routing Slip Number (9 digits)"
+            :value="routingSlipNumber"
+            :rules="rules"
+            :disabled="paymentOption !== PAYMENT_RECEIVED"
+            @input="emitRoutingSlipNumber($event)"
+          />
+          <v-checkbox
+            id="priority-checkbox"
+            label="Priority (add $100.00)"
+            :input-value="isPriority"
+            :disabled="paymentOption !== PAYMENT_RECEIVED"
+            @change="emitIsPriority(!!$event)"
+          />
+        </v-form>
+
+        <!-- No Fee radio button -->
+        <v-radio id="no-fee-radio" label="No Fee" />
+      </v-radio-group>
+    </div>
+  </v-card>
+</template>
+
+<script lang="ts">
+import { Component, Vue, Prop, Watch, Emit } from 'vue-property-decorator'
+
+@Component({})
+export default class StaffPayment extends Vue {
+  // payment options:
+  readonly PAYMENT_RECEIVED = 0
+  readonly NO_FEE = 1
+
+  /** Routing Slip Number prop. */
+  @Prop({ default: '' })
+  private routingSlipNumber: string
+
+  /** Is Priority prop. */
+  @Prop({ default: false })
+  private isPriority: boolean
+
+  /** Is Waive Fees prop. */
+  @Prop({ default: false })
+  private isWaiveFees: boolean
+
+  /** Form model property. */
+  private formValid: boolean = false
+
+  /** Array of validation rules for Routing Slip Number. */
+  private readonly rules = [
+    v => !!v || 'Routing Slip Number is required',
+    v => /^\d{9}$/.test(v) || 'Routing Slip Number must be 9 digits'
+  ]
```

Comment:
Do these rules exist elsewhere as well? I can't recall?

Reply:
Yes, different components have different rules for different properties. Also the validation error messages are different for different fields.

Reply:
Ahh i see, yes the msg's are closely tied to the rules and are very specific. Makes sense they are where they are!